Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject: deadwood source code corrupted during build
 Also we don't like binaries with a capital in the name.
Forwarded: not-needed
Last-Update: 2011-09-17
--- a/deadwood-3.0.05/src/Makefile
+++ b/deadwood-3.0.05/src/Makefile
@@ -20,7 +20,7 @@
 	DwRecurse.o \
 	DwDict.o
 
-all:	Deadwood version.h
+all:	deadwood version.h
 
 # Since some systems may not have /dev/urandom (Windows, *cough* *cough*), we 
 # keep a randomly generated prime around 
@@ -30,12 +30,8 @@
 
 clean:
 	rm -f Test DwMain DwTcp *.exe *.o a.out RandomPrime writehash_test* \
-		Deadwood foo* dw_cache DwHash DwCompress *stackdump \
-		core ; \
-		./make.version.h ; if [ -e /dev/urandom ] ; \
-			then rm DwRandPrime.h  ; \
-			cc RandomPrime.c ; ./a.out > DwRandPrime.h ; rm a.out \
-		; fi 
+		deadwood foo* dw_cache DwHash DwCompress *stackdump core
+		if [ -f DwRandPrime.h.bak ]; then mv DwRandPrime.h.bak DwRandPrime.h; fi
 
 version.h:	
 	./make.version.h
@@ -71,7 +67,7 @@
 	$(CC) -O3 -o RandomPrime RandomPrime.c
 
 DwRandPrime.h: RandomPrime
-	if [ -e /dev/urandom ] ; then ./RandomPrime > DwRandPrime.h ; fi
+	if [ -e /dev/urandom -a -f DwRandPrime.h ] ; then mv -f DwRandPrime.h DwRandPrime.h.bak ; ./RandomPrime > DwRandPrime.h ; fi
 
 DwHash.o:	DwHash.c DwStr.h DwRandPrime.h DwHash.h
 	$(CC) $(FLAGS) -Wall -c -o DwHash.o DwHash.c
@@ -85,6 +81,6 @@
 Test:		Test.c DwStr.o DwStr.h DwStr_functions.h $(OBJS)
 	$(CC) $(FLAGS) -Wall -o Test Test.c $(OBJS)
 
-Deadwood:	DwMain.c $(OBJS) DwStr_functions.h version.h
-	$(CC) $(FLAGS) -Wall -o Deadwood DwMain.c $(OBJS) 
+deadwood:	DwMain.c $(OBJS) DwStr_functions.h version.h
+	$(CC) $(FLAGS) -Wall -o deadwood DwMain.c $(OBJS)
 
